packages/datatrak-web-server/src/routes/TasksRoute.ts

```diff
@@ -161,7 +154,20 @@ export class TasksRoute extends Route<TasksRequest> {
       page,
     });
 
-    const formattedTasks = tasks.map((task: TaskT) => formatTaskResponse(task));
+    const formattedTasks = (await Promise.all(
+      tasks.map(async (task: TaskT) => {
+        const formattedTask = formatTaskResponse(task);
+        // Get comment count for each task
+        const commentsCount = await models.taskComment.count({
```

Comment:
I couldn't think of a good way to fetch tasks outside of this. I tried a custom column selector but I don't think that works for counts from another joined table (because to have a `count` you also need to put it into an aggregate function/group by etc) so this seemed simpler.
I also didn't want to fetch the tasks from the db directly, because the central server does some nice permissions handling for us

Reply:
All good